### DISCUSSION TITLE
strongswan-5.4 auto=route issue with overlapping dst subnets

### DISCUSSION THREAD

**Discussion by mmabln** [strongswan/strongswan#793] [Q&A]
Hi, I've a problem with a config like this:

tunnel_1: right=1.2.3.4, leftsubnet=10.200.0.0/28, rightsubnet=10.198.0.0/16
tunnel_2: right=5.6.7.8, leftsubnet=10.200.0.0/28, rightsubnet=10.198.64.0/24

Both tunnels are configured with auto=route. When tunnel 1 is up and tunnel 2 is down, traffic to 10.198.64.0/24 is not establishing tunnel 2 (which was my expectation), but traffic is sent to tunnel 1 instead.

Once tunnel 2 is established with "strongswan up tunnel_2", traffic with dst 10.198.64.0/24 is routed correctly to tunnel 2.

The XFRM policies look like this, so is this the problem? How to fix?

src 10.200.0.0/28 dst 10.198.0.0/16
dir out priority 2899 ptype main
tmpl src xxx dst 1.2.3.4
proto esp reqid 15 mode tunnel

src 10.200.0.0/28 dst 10.198.0.0/16
dir out priority 5971 ptype main
tmpl src xxx dst 1.2.3.4
proto esp reqid 15 mode tunnel

src 10.200.0.0/28 dst 10.198.64.0/24
dir out priority 5939 ptype main
tmpl src xxx dst 5.6.7.8
proto esp reqid 2 mode tunnel

Thanks & regards Markus

**Comment by tobiasbrunner** [comment]
Please try a newer version (at least [5.5.2](https://wiki.strongswan.org/versions/64)).
